account_cash_discount_payment/wizards/account_payment_line_create.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+
+
+class AccountPaymentLineCreate(models.TransientModel):
+
+    _inherit = 'account.payment.line.create'
+
+    date_type = fields.Selection(
+        selection_add=[
+            ('discount_due_date', _("Discount Due Date")),
+        ],
+    )
+    cash_discount_date_start = fields.Date()
+    cash_discount_date_end = fields.Date()
```

Comment:
Do we need start + end? For due date, the test is `('date_maturity', '<=', self.due_date)`, I think we can do the same for discount due date, it's easier and it's the most common use case.

Reply:
I don't know if start/end range is really needed, we should ask @chrisdec. But in my opinion, we could have a single date indeed.